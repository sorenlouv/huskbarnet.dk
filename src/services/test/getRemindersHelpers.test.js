import dbResponse from './response.json';
import {
  formatReminders,
  getEmailsToSend,
  getDate
} from '../getRemindersHelpers';

describe('formatReminders', () => {
  it('should format reminders correctly', () => {
    expect(formatReminders(dbResponse)).toMatchSnapshot();
  });
});

describe('getDate', () => {
  it('should return correct dates', () => {
    expect(getDate('2015-04-01', 30)).toBe('2015-03-02');
  });
});

describe('getEmailsToSend', () => {
  it('It should format messages correctly', () => {
    const adminRefMock = {};
    adminRefMock.database = () => adminRefMock;
    adminRefMock.ref = () => adminRefMock;
    adminRefMock.orderByChild = () => adminRefMock;
    adminRefMock.equalTo = () => adminRefMock;
    adminRefMock.once = () => adminRefMock;
    adminRefMock.then = cb => cb({ val: () => dbResponse });

    return expect(
      getEmailsToSend('2017-12-22', adminRefMock)
    ).resolves.toMatchSnapshot();
  });
});
