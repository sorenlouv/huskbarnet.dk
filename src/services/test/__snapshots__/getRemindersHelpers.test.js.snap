// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatReminders should format reminders correctly 1`] = `
Array [
  Object {
    "dateOfBirth": "2017-12-22",
    "email": "larsfar@gmail.com",
    "emails": Object {
      "auth": "larsfar@gmail.com",
    },
    "gender": "male",
    "name": "Lars",
  },
  Object {
    "dateOfBirth": "2016-08-15",
    "email": "sillemor@gmail.com",
    "emails": Object {
      "auth": "sillemor@gmail.com",
    },
    "gender": "female",
    "name": "Sille",
  },
]
`;

exports[`getEmailsToSend It should format messages correctly 1`] = `
Array [
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 3 måneders vaccination hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 3 måneders vaccination hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 5 måneders vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 5 måneders vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 12 måneders vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 12 måneders vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 15 måneders vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 15 måneders vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 4 års vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 4 års vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 5 års vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 5 års vaccination og til børneundersøgelse hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om børneundersøgelserne på https://www.sundhed.dk/borger/patienthaandbogen/boern/undersoegelser/boerneundersoegelser/
Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination og børneundersøgelse af Sille",
  },
  Object {
    "email": "larsfar@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Lars skal have sin 12 års vaccination hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination af Lars",
  },
  Object {
    "email": "sillemor@gmail.com",
    "message": "Kære forælder,

Dette er en påmindelse om, at dit barn Sille skal have sin 12 års vaccination hos jeres læge. Du skal selv huske at bestille tid.

Læs mere om det danske vaccinations program på https://sst.dk/da/sygdom-og-behandling/vaccinationer/~/media/811A9F6CD64B4462B6FDFE503787CC71.ashx

Med venlig hilsen
BørnePåmindelser.dk",
    "subject": "Påmindelse om vaccination af Sille",
  },
]
`;
